template_feed/Microsoft.DotNet.Common.ItemTemplates/content/Class-CSharp/.template.config/template.json

```diff
@@ -0,0 +1,75 @@
+{
+  "$schema": "http://json.schemastore.org/template",
+  "author": "Microsoft",
+  "classifications": [
+    "Common"
+  ],
+  "name": "Class",
+  "generatorVersions": "[1.0.0.0-*)",
+  "description": "A template for creating a class item",
+  "tags": {
+    "language": "C#",
+    "type": "item"
+  },
+  "groupIdentity": "DotNet.Common.Class",
+  "precedence": "0",
+  "identity": "DotNet.Common.Class.CSharp",
+  "shortName": "class",
+  "sourceName": "Class1",
+  "primaryOutputs": [
+    {
+      "path": "Class1.cs"
+    }
+  ],
+  "symbols": {
+    "DefaultNamespace": {
+      "type": "bind",
+      "binding": "msbuild:RootNamespace",
+      "replaces": "Company.ClassLibrary1"
+    },
+    "TargetFramework": {
+      "type": "bind",
+      "binding": "msbuild:TargetFramework"
+    },
+    "net6orLater": {
+      "type": "generated",
+      "generator": "regexMatch",
+      "datatype": "bool",
+      "parameters": {
+        "pattern": "^(net6.0|net7.0|net8.0)$",
+        "source": "TargetFramework"
+      }
+    },
+    "langVersion": {
+      "type": "bind",
+      "binding": "msbuild:LangVersion"
+    },
+    "csharp9orOlder": {
+      "type": "generated",
+      "generator": "regexMatch",
+      "datatype": "bool",
+      "parameters": {
+        "pattern": "^(ISO-1|ISO-2|[1-7]|[8-9]|[8-9]\\.0|7\\.[0-3])$",
+        "source": "langVersion"
+      }
+    },
+    "csharp10orLater": {
+      "type": "computed",
+      "value": "!csharp9orOlder"
+    },
+    "csharpFeature_ImplicitUsings": {
+      "type": "computed",
+      "value": "net6orLater == \"true\" && csharp10orLater == \"true\""
+    },
+    "csharpFeature_FileScopedNamespaces": {
+      "type": "computed",
+      "value": "(net6orLater == \"true\" || langVersion != \"\") && csharp10orLater == \"true\""
+    }
+  },
```

Comment:
Unfortunately there's no matching MSBuild property for this one :) It could maybe be replaced with a property-binding as described above, however: `$([MSBuild]::VersionGreaterThanOrEquals($(LangVersion), '10.0'))`

However, because this is a _style_ question, we probably need to defer until we can use dotnet-format to format the code.

Alternative feature idea: `bind` syntax for `editorconfig`?